### PR TITLE
[docs]close #12580

### LINKED ISSUE #12580: Confusing pairs create/dealloc, createShared/freeShared
Looking at the current system documentation the typed alloc/free pairs are confusing:

### alloc/create --> dealloc

https://github.com/nim-lang/Nim/blob/4e0f12092ebb64047dc0ecfc86e348e4d93eab4c/lib/system.nim#L2397-L2423

### dealloc <--> alloc

https://github.com/nim-lang/Nim/blob/4e0f12092ebb64047dc0ecfc86e348e4d93eab4c/lib/system.nim#L2473-L2483

### allocShared <--> deallocShared

https://github.com/nim-lang/Nim/blob/4e0f12092ebb64047dc0ecfc86e348e4d93eab4c/lib/system.nim#L2485-L2497

### createShared <--> freeShared

https://github.com/nim-lang/Nim/blob/4e0f12092ebb64047dc0ecfc86e348e4d93eab4c/lib/system.nim#L2525-L2536

## The confusion

`create` does not have a `free` to pair with, unlike `createShared`.
Also while `create` refers to `dealloc` in the documentation, `dealloc` only refers to `alloc`.

### PR DESCRIPTION
**Pull request by ringabout**
close #12580
mention `create` at the documentation of `dealloc`.

### CODE REVIEW

File: lib/system/memalloc.nim [hunk 1]
```diff
@@ -188,8 +188,8 @@ when hasAlloc and not defined(js):
     cast[ptr T](realloc(p, T.sizeof * newSize))
 
   proc dealloc*(p: pointer) {.noconv, compilerproc, rtl, benign, raises: [], tags: [].} =
-    ## Frees the memory allocated with `alloc`, `alloc0` or
-    ## `realloc`.
+    ## Frees the memory allocated with `alloc`, `alloc0`,
+    ## `realloc` or `create`.
```
Comment (konsumlamm) [inline lib/system/memalloc.nim]: Maybe link those?

Comment (timotheecour) [reply lib/system/memalloc.nim]: ok to defer IMO; let's just implement https://github.com/nim-lang/RFCs/issues/125 instead; then you'll just write
```
realloc_ or create_.
```

File: lib/system/memalloc.nim [hunk 2]
```diff
@@ -188,8 +188,8 @@ when hasAlloc and not defined(js):
...
     ##
```
Comment (timotheecour) [inline lib/system/memalloc.nim:193]: what about `createU`?

File: lib/system/memalloc.nim (hunk 1 above)
Comment (ringabout) [inline lib/system/memalloc.nim]:
```suggestion
    ## `realloc`, `create` or `createU`.
```

File: lib/system/memalloc.nim (hunk 2 above)
Comment (ringabout) [reply lib/system/memalloc.nim:193]: thanks

**Review by timotheecour — APPROVED** [review]

File: lib/system/memalloc.nim (hunk 1 above)
Comment (ringabout) [reply lib/system/memalloc.nim]: make sense

**Pull request merged**